File: 2616-maximal-score-after-applying-k-operations/maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        long score = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>(Collections.reverseOrder());
        for (int i: nums)
            pq.add(i);
        while(k > 0){
            int top = pq.poll();
            score += top;
            pq.add((int)(Math.ceil(top / 3.0)));
            k -= 1;
        }
        return score;
    }
}

File: 2616-maximal-score-after-applying-k-operations/maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        long long score = 0;
        priority_queue<int> pq;
        for (int i: nums)
            pq.push(i);
        while(k > 0){
            int top = pq.top();
            pq.pop();
            score += top;
            pq.push(ceil(top / 3.0));
            k -= 1;
        }
        return score;
    }
};